features/project/project_controller.rb

```diff
@@ -116,8 +116,19 @@ class ProjectController < AuthenticatedControllerBase
       lane = params["selected_lane"]
       project_name = params["project_name"]
 
+      # Do this so we trigger the clone of the repo.
+      # TODO: Do this wherever it should be done, as we must redirect
+      # to the project details only when this task is finished.
+      _ = GitRepo.new(
```

Comment:
This should be moved into a service

Reply:
I'm really thinking that in a follow up PR, refactoring all the logic between project's repos and ci-repo itself. They use the same underneath logic but their use cases are totally different. Bot I think is a little bit out of the scope of this PR, What do you think?

Reply:
I think that's a good idea. That being said, this shouldn't exist here 😄 controllers are meant to handle display logic, and to call into services which are meant to handle business logic. Mixing display with business logic, and in this case, relying on a side-effect to perform some business logic is extra dangerous.

Reply:
May I cover this in a separate PR as it involves pretty big refactoring?

Reply:
Yup, I just wouldn't want this to stick around and be forgotten 😄

Reply:
Written #280 in order to tackle this soon!

Reply:
👍 thanks